extension/src/ohlc.rs

```diff
@@ -0,0 +1,644 @@
+use pgx::*;
+
+use crate::{
+    aggregate_utils::in_aggregate_context,
+    flatten,
+    palloc::{Inner, Internal, InternalAsValue, ToInternal},
+    pg_type,
+    raw::bytea,
+    ron_inout_funcs,
+};
+use tspoint::TSPoint;
+
+#[pg_schema]
+pub mod toolkit_experimental {
+    use super::*;
+
+    pg_type! {
+        #[derive(Copy, Debug)]
+        #[repr(C)]
+        struct OpenHighLowClose {
+            open: TSPoint,
+            high: TSPoint,
+            low: TSPoint,
+            close: TSPoint,
+        }
+    }
+
+    impl OpenHighLowClose<'_> {
+        pub fn new() -> Self {
```

Comment:
As discussed, probably better if this takes the first point.  Feel free to add a comment and file an issue to address this later.